Clarify site parts for footers (#2128)
jupyter-book/mystmd: docs/document-parts.md

description: Parts allow you to specify special parts of your document, like abstract, key points, and acknowledgements.
---

> Document parts allow you to add metadata to your documents with specific components of your page or project, for example, abstract, dedication, or acknowledgments. [Templates](./documents-exports.md) can use this information to put that content in various locations.

Reviewer: ```suggestion
Document *parts* allow you to add content and metadata for specific components of your page or project, including `example`, `abstract`, `dedication`, and `acknowledgment`. [Templates](./documents-exports.md) may use the parts and their metadata, and render such content in various locations.
```

## Add document parts

There are several ways you can define parts of a document, each described below:

> 1. [In page frontmatter](#parts:frontmatter)
> 2. [In project-level configuration](#parts:project)
> 3. [With specific section headings](#parts:implicit)
> 4. [With a content block](#parts:blocks)
> 5. [With Jupyter Notebook cell tags](#parts:cell-tags)
> 6. [In site configuration](#parts:site)

Reviewer: ```suggestion
1. [In site configuration](#parts:site)
2. [In project-level configuration](#parts:project)
3. [In page frontmatter](#parts:frontmatter)
4. [With specific section headings](#parts:implicit)
5. [With a content block](#parts:blocks)
6. [With Jupyter Notebook cell tags](#parts:cell-tags)
```

Reviewer: Order by scope.

---
```

> ### Known Frontmatter Parts

Reviewer 2: Where is this getting moved to?

Reviewer 2: We could leave this in the reference side?

Author: Good catch, definitely an accident!

---
```

(parts:project)=

> ### Parts in `myst.yml` Project configuration

Reviewer: See sorting suggestion above.

The advantage of this method is that the content is not rendered in your document.

(parts:site)=

## Add parts to your website

> [Website themes](./website-templates.md) have their own configuration for parts because they often have extra user interface elements that are not part of a standard "document" structure. These are theme-dependent. For example, the [default myst themes](#default-web-themes) support [a website footer](#navigation:footer).

Reviewer: ```suggestion
[Website themes](./website-templates.md) have additional parts because they render user interface elements that are not part of a standard "document" structure. These are theme-dependent: for example, the [default myst themes](#default-web-themes) support a [`footer` part](#navigation:footer).
```

Specify the content of a site `part` in the `site.parts` key of the `myst.yml` configuration file:

```{code} yaml
:filename: myst.yml
version: 1
site:
  template: ...
  parts:
    footer: |
      (c) MyST Markdown
  ...
```

Alternatively, you may specify a path to a file that contains the part content:

```{code} yaml
:filename: myst.yml
version: 1
site:
  template: ...
  parts:
    footer: parts/myfooter.md
  ...
```

Content in parts will generally be parsed similarly to other MyST content (though some functionality like code execution will not work).

### Share the same part across multiple sites

> If you wish to share the same part across multiple sites, use the [`extends:` key to compose multiple configuration files](#composing-myst-yml). This lets you define the part in one location, and re-use it in several others. It is useful if you want to standardize website components like a footer across many websites.

Reviewer: How does composing configuration files help here? E.g., is it easier to pull in the config that pulls in the part? Or can a person pull the part in directly? (Maybe not possible?)

Reviewer: But wouldn't you just want to point to that *part* in your current site config?

Author: I don't think that is possible if the part isn't local, at least until this enhancement is made:

- https://github.com/jupyter-book/mystmd/issues/2127
